lighthouse/datadog_checks/lighthouse/lighthouse.py

```diff
@@ -0,0 +1,83 @@
+# the following try/except block will make the custom check compatible with any Agent version
+try:
+    # first, try to import the base class from old versions of the Agent...
+    from checks import AgentCheck
+except ImportError:
+    # ...if the above failed, the check is running in Agent version 6 or later
+    from datadog_checks.checks import AgentCheck
+    from datadog_checks.errors import CheckException
+
+from datadog_checks.utils.subprocess_output import get_subprocess_output
+import json
+
+# content of the special variable __version__ will be shown in the Agent status page
+__version__ = "1.0.0"
+
+EXPECTED_RESPONSE_CODE = "NO_ERROR"
+
+
+class LighthouseCheck(AgentCheck):
+    def check(self, instance):
+        lighthouse_url = instance.get('url')
+        lighthouse_name = instance.get('name')
+
+        if not lighthouse_url or not lighthouse_name:
+            self.log.error("missing instance url or name")
+            raise CheckException("missing lighthouse instance url or name, please fix yaml")
+
+        cmd = ["/usr/local/bin/lighthouse",
+               lighthouse_url,
+               "--output",
+               "json",
+               "--quiet",
+               "--chrome-flags='--headless'"]
+
+        json_string, error_message, exit_code = self._get_lighthouse_report(cmd, self.log, False)
+
+        # check for error since we have raise_on_empty_output set to False
+        if exit_code > 0:
+            self.log.error("lighthouse subprocess error {0} exit code {1} for url: {2}"
+                           .format(error_message, exit_code, lighthouse_url)
+                           )
+            raise CheckException(json_string, error_message, exit_code)
+
+        try:
+            data = json.loads(json_string)
+        except Exception as e:
+            self.log.warn("lighthouse response JSON different than expected for url: {0}".format(lighthouse_url))
+            raise CheckException(error_message, exit_code, e)
+
+        if data["runtimeError"]["code"] == EXPECTED_RESPONSE_CODE:
+            score_accessibility = round(data["categories"]["accessibility"]["score"] * 100)
+            score_best_practices = round(data["categories"]["best-practices"]["score"] * 100)
+            score_performance = round(data["categories"]["performance"]["score"] * 100)
+            score_pwa = round(data["categories"]["pwa"]["score"] * 100)
+            score_seo = round(data["categories"]["seo"]["score"] * 100)
```

Comment:
This may be due to my level of python knowledge, but I was running into issues where the scores were being represented as, for example, 57.000000000000001, or something along those lines. So the rounding way my hacky way of fixing that. Fwiw these scores are only represented as integers, so in practice it shouldn't impact any of the data precision.

If there's a more pythonic way of handling this I would be happy to change it, open to suggestions and appreciate you taking a look.

Reply:
> This may be due to my level of python knowledge, but I was running into issues where the scores were being represented as, for example, 57.000000000000001, or something along those lines.

This is due to the way python (same for many other languages) represent floats, some decimal values cannot be represented exactly. Here are few links, if you want to dig into this :) https://docs.python.org/3/tutorial/floatingpoint.html, see also IEEE-754.

> Fwiw these scores are only represented as integers, so in practice it shouldn't impact any of the data precision.

Should be fine, in that case.